### ISSUE TITLE
Terrain (hillshade and slope) never visible

### ISSUE THREAD

**Issue by uhkeller** [osmandapp/OsmAnd#18179]
### Describe the bug (required)

The settings in Configure map -> Terrain -> Terrain have no effect whatsoever in my current device. Neither slope not hillshade are shown, even at 100% visibility. The zoom level makes no difference.
The terrain maps (3D) are downloaded. I'm using the v2 (OpenGL) renderer. Restarting the app and clearing the cache have no effect. I've used OsmAnd+ on a different device until two weeks ago (a Fairphone 3) and hillshade and slope rendering worked just fine.

https://github.com/osmandapp/OsmAnd/assets/2323704/e1be9830-dd0b-4e19-96f7-3a8d1223a344

### Describe how the bug can be reproduced (required)

On a Samsung A54 with a downloaded terrain map, got to Configure map -> Terrain -> Terrain and enable slope or hillshade rendering.

### Describe the actual result (required)

No slope or hillshade rendering.

### Describe the expected result (required)

Slope or hillshade rendering.

### Your Environment (required)

**WARNING** Crash-Logs **MAY** contain information you deem sensitive.
Review this **CAREFULLY** before posting your issue!

```
OsmAnd Version: 4.5.10
Android/iOS version: Android 13
Device model: Samsung Galaxy A54 5G
Crash-Logs: ?
```

**Comment by Spielmops** [comment]
Wish: I would really like to get a little clarification on this topic: what has changed with the introduction of the OpenGL driver, why there are 2 maps (relief and terrain), what works with what, which switches do what.

Spielmops

**Comment by dmpr0** [comment]
@Spielmops 
Hi! You can read about it on our blog https://osmand.net/blog/terrain

**Comment by Spielmops** [comment]
I know that already, but how does that work with renderinghengine 1? I have 2 tablets, one both of them the same installation of Osmand+ and plugins. But one works with OpenGL and the other not. I am not able to get a relief ...

**Comment by uhkeller** [comment]
I tried:

* Moving the data storage folder from the SD card to internal storage
* Deleting the terrain maps and redownloading them through the Terrain settings

No change. Terrain stays invisible. I'm of course willing to supply any log data that might help debugging this.

**Comment by DmitryAlexei** [comment]
@uhkeller could you please check:
- Is *use raster sqlitedb for hillshade / slope* option in OsmAnd Development plugin settings (menu > plugins > OsmAnd development) switched on or off? (this option is used for old hillshades)
- if you turn this option on and download raster hillshade maps, are hillshades visible?
Do you use OsmAnd from Gplay or Fdroid? You can also join beta testing and check if this option works on your device in beta

**Comment by uhkeller** [comment]
Thanks @DmitryAlexei. The option is off, but greyed out and can't be enabled.
I'm using OsmAnd+ from Google Play (paid).

**Comment by DmitryAlexei** [comment]
So this problem has not been solved in https://github.com/osmandapp/OsmAnd/issues/17783

**Comment by uhkeller** [comment]
Looks like it. Though as I mentioned, it *did* work on my previous phone (Fairphone 3) with the same version of OsmAnd.

**Comment by Spielmops** [comment]
It did work on my tablet (Samsung S2) with previous version of Osmand (paid).

Spielmops

**Comment by Spielmops** [comment]
What I tried with my older tablet:

I installed Osmand 4.2.7, that is imho the last version without OpenGL. I downloaded the map "relief", that is what made the hillshade in earlier versions. Result: no hillshade! 

If I set Osmand to show the hillshade-map as a map-source, nothimg is shown. And not as an overlay-map. 

So my conclusion: the hillshade-map is faulty ...

Spielmops

**Comment by uhkeller** [comment]
@Spielmops, I intended this issue to be about hillshade/slope with the new renderer based on the new 3D terrain maps. Maybe there should be a different issue for you trouble with the old renderer and the relief maps?

Anyway, I joined the beta program and am now using 4.6.1. With this version slopes render fine, but hillshades look ok only in high zoom levels. If I zoom out the area with hillshade is confined more and more to the left and lower edge of the map (see screenshots).
![Screenshot_20230923_134752_OsmAnd+](https://github.com/osmandapp/OsmAnd/assets/2323704/e8e3f782-45b9-4747-b659-c35176af3c1b)
![Screenshot_20230923_134806_OsmAnd+](https://github.com/osmandapp/OsmAnd/assets/2323704/f84a004f-ff43-4993-8e7d-daf1c98843fa)

**Comment by DmitryAlexei** [comment]
**OsmAnd 4.5.10, released: 2023-08-09**
Steps to reproduce:
- go to menu > Download maps > All downloads > Europe > Austria and download 3d map

Actual result:
- **new hillshades are not seen on the map**
- raster hillshades are visible when *Use raster sqlitedb for hillshade / slope* is enabled in development plugin and old hillshade map is downloaded


<table style="width:100%">
  <tr>
    <th>Maps downloaded</th>
    <th>Hillshade settings</th>
    <th>Map view</th>
    <th>Old hillshades with the same settings</th>
  </tr>
  <tr>
<th><img src="https://github.com/osmandapp/OsmAnd/assets/116079541/d46637ab-3463-4499-841c-d5cc2bfccd4b"  width="300" height="auto"/> </th>
 <th><img src="https://github.com/osmandapp/OsmAnd/assets/116079541/35745a0e-b7de-43ac-ac86-61eb5e01d4cf"  width="300" height="auto"/> </th>
 <th><img src="https://github.com/osmandapp/OsmAnd/assets/116079541/4b4ee5a7-abe9-421c-9170-7a9511b2cc09"  width="300" height="auto"/> </th>
 <th><img src="https://github.com/osmandapp/OsmAnd/assets/116079541/1cd32da1-879c-40a1-a30f-c56367208017"  width="300" height="auto"/> </td>
  </tr>
</table>

**This bug is fixed in OsmAnd~ 4.6.0#806m, released: 2023-09-25**
Actual result:
- new hillshades are visible on the map
- raster hillshades are visible when *Use raster sqlitedb for hillshade / slope* option is enabled in development plugin and old hillshade map is downloaded
- new hillshades are not visible only when *Use raster sqlitedb for hillshade / slope* option is enabled and old hillshade maps are not downloaded (as expected)

<table style="width:100%">
  <tr>
    <th>Maps downloaded</th>
    <th>New hillshades</th>
    <th>Old hillshades with the same settings</th>
  </tr>
  <tr>
<th><img src="https://github.com/osmandapp/OsmAnd/assets/116079541/2e36fdd3-7833-4860-9692-0853e9e5a94f"  width="300" height="auto"/> </th>
 <th><img src="https://github.com/osmandapp/OsmAnd/assets/116079541/518d7e70-77a7-412f-b0cd-169e62627a28"  width="300" height="auto"/> </th>
 <th><img src="https://github.com/osmandapp/OsmAnd/assets/116079541/48e1443f-dd75-49b8-86ab-21056de839c7"  width="300" height="auto"/> </th>
  </tr>
</table>

**Comment by vshcherb** [comment]
You probably need to delete old raster data  if you have and double check development settings
